### PR TITLE
Use `PyYAML==5.1` on `tests-with-minimum-dependencies`

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Refs #4428 

Resolve an installation error of PyYAML.

## Description of the changes

It's failed to install `PyYAML==3.1.0` at:
https://github.com/optuna/optuna/actions/runs/4201205273/jobs/7287987185

### CODE REVIEW

File: .github/workflows/tests-with-minimum-versions.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+  pull_request: {}
```
Comment (c-bata) [inline .github/workflows/tests-with-minimum-versions.yml:7]: TODO: Remove this temporary change before merging.

```suggestion
```

Comment (c-bata) [reply .github/workflows/tests-with-minimum-versions.yml:7]: I confirmed that this PR resolves the error.
https://github.com/optuna/optuna/actions/runs/4201654625/jobs/7288874421

**Comment by c-bata** [comment]
@contramundum53 @Alnusjaponica  Could you review this PR?

**Comment by Alnusjaponica** [comment]
> failed to install PyYAML==3.1.0

I wonder if it is better to introduce the minimum required version of PyYAML in pyproject.toml instead of just giving up on testing.

**Comment by c-bata** [comment]
Sounds reasonable. This problem does not block to merge other PRs, so I'll investigate the minimum required version of PyYAML.

**Comment by c-bata** [comment]
PyYAML 3.13, released at July 6th, 2018, seems to be the minimum version. I've pushed [b9d16b3](https://github.com/optuna/optuna/commit/b9d16b30abf538434dc21470ca7a8779955d7016).

```
$ docker run -it --rm --name optuna-pyyaml python:3.7 bash
root@3338a4fb301f:/# pip install PyYAML==3.13
Collecting PyYAML==3.13
  Downloading PyYAML-3.13.tar.gz (270 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 270.6/270.6 KB 7.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: PyYAML
  Building wheel for PyYAML (setup.py) ... done
  Created wheel for PyYAML: filename=PyYAML-3.13-cp37-cp37m-linux_aarch64.whl size=498619 sha256=776765eed66c51c2575adea0201b650945ce9a8017ca3e26601c6f2fc68e4709
  Stored in directory: /root/.cache/pip/wheels/95/cd/14/899edaa9cdb9a65aa7224539f6e0ad488e9a7b202bb48f6ae6
Successfully built PyYAML
Installing collected packages: PyYAML
Successfully installed PyYAML-3.13
```

```
$ docker run -it --rm --name optuna-pyyaml python:3.7 bash
root@3338a4fb301f:/# pip install PyYAML==3.12
      ext/_yaml.c:24217:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
      24217 |     tstate->exc_traceback = local_tb;
            |             ^~~~~~~~~~~~~
            |             curexc_traceback
      error: command 'gcc' failed with exit status 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> PyYAML
```

https://pypi.org/project/PyYAML/#history

**Review by Alnusjaponica — APPROVED** [review]
Thank you for your detailed investigation. LGTM.

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Pull request merged**